packages/grid/_modules_/grid/components/GridRowCount.tsx

```diff
@@ -25,7 +27,11 @@ export const GridRowCount = React.forwardRef<HTMLDivElement, GridRowCountProps>(
         : rowCount.toLocaleString();
 
     return (
-      <div ref={ref} className={clsx(gridClasses.rowCount, className)} {...other}>
+      <div
+        ref={ref}
+        className={clsx(gridClasses.rowCount, rootProps.classes?.rowCount, className)}
```

Comment:
Is it impossible to just process the `class` prop and have `rootProps.classes?.XXX` always contains `gridClasses.XXX` and `inputProps.classes?.XXX` ?

To avoid all these merge in the hooks / components.

Reply:
It's possible, but this same pattern is used in the core. They chose to be verbose.

Reply:
In the core, I believe we do as Flavien suggests https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/Alert/Alert.js#L167, composeClasses merge the global classes with the classes prop.

Reply:
From what I understood, https://github.com/mui-org/material-ui-x/pull/2458#discussion_r698295681 is proposing to merge in a single place (e.g. useDataGridProps) the `classes` that the user is providing with the default grid's classes. By doing that, `composeClasses` will include each class twice:

`<DataGrid classes={{ cell: 'foobar' }} />`

`rootProps.classes.cell` -> `"MuiDataGrid-cell foobar"`

`composeClasses(['cell'], ..., rootProps.classes)` -> `"MuiDataGrid-cell MuiDataGrid-cell foobar"`

The alternative is to implement the `useUtilityClasses` pattern in all the places.

Reply:
Yes my proposal is what @m4theushw is describing.
If it can be done without downsides, I think it's cleaner.
But if it duplicates data, it may not be a good idea.